docs/features/search/search-for-plugin-devs.md

```diff
@@ -0,0 +1,187 @@
+---
+id: for-plugin-devs
+title: Search for Plugin Developers
+description: How to integrate search into a Backstage plugin
+---
+
+The Backstage Search Platform was designed to give plugin developers the APIs
+and interfaces needed to offer search experiences within their plugins, while
+abstracting away (and instead empowering application integrators to choose) the
+specific underlying search technologies.
+
+On this page, you'll find concepts, guides, and recipes for how to leverage the
+Backstage Search Platform in your plugin.
+
+## Providing data to the search platform
+
+> A guide on how to create collators is coming soon!
+
+## Building a search experience into your plugin
+
+While the core Search plugin offers components and extensions that empower app
+integrators to compose a global search experience, you may find that you want a
+narrower search experience just within your plugin. This could be as literal as
+an autocomplete-style search bar focused on documents provided by your plugin,
+or as abstract as a widget that presents a list of links that relate in some
+way to something else on the page.
+
+### Concepts
+
+Knowing these high-level concepts will help you as you craft your in-plugin
+search experience.
+
+- All search experiences must be wrapped in a `<SearchContextProvider>`, which
+  is provided by `@backstage/plugin-search-react`. This context keeps track
+  of state necessary to perform search queries and display any results. As
+  inputs to the query are updated (e.g. a `term` or `filter` values), the
```

Comment:
I think it would be nice to specify all state (or link to) that I've access to on the context here.

Reply:
Also a good idea.  Added here: https://github.com/backstage/backstage/pull/11675/commits/9387626eeccef35f5a611aa256a5d6d02108e6c1